Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -1207,6 +1217,7 @@ void WebProcessProxy::didFinishLaunching(ProcessLauncher* launcher, IPC::Connect
 #endif // USE(RUNNINGBOARD)
 
     m_throttler.setShouldTakeSuspendedAssertion(shouldTakeSuspendedAssertion());
```

Comment:
I think these two settings are too confusing with this proposal. It seems like not taking suspended assertions should be identical to dropping suspended assertions immediately.

My proposal:
- shouldTakeSuspendedAssertion setting:
  * ON: Takes a Suspended assertion when the process throttler goes to suspended state (current behavior)
  * OFF: Do not take any assertions when the process throttler goes to suspended state (new behavior, basically what you're doing in this patch behind another setting.

- shouldDropSuspendedAssertionAfterDelay:
  * ON: Takes a Suspended assertion when the process throttler goes to suspended state and drops the assertion when the timer fires (new behavior, This used to be the shouldTakeSuspendedAssertion=OFF behavior)
  * OFF: Don't start a timer to drop the Suspended assertion when the process throttler goes to suspended state